### DISCUSSION TITLE
Question about network interface at runtime

### DISCUSSION THREAD

**Discussion by advancenXX** [eProsima/Fast-DDS#3510] [Q&A]
I am aware of the Dynamic-network-interface feature so that I can invoke set_qos() when a new network interface is enabled.
But I am not sure if it can work well as it is still under development.

And I am also wondering if it is possible to pass a STATIC IP to fastdds before this network interface is enabled. My network interface is fixed so that I want to set it to fastdds at an earlier time.

**Comment by JLBuenoLopez** [comment]
Dynamic network interfaces work for UDPv4. There is a [known issue](https://github.com/eProsima/Fast-DDS/issues/2676#issuecomment-1134625889) that if initially no network interface is enabled, even if a network interface is included afterwards, communication is not established.

Concerning the static IP address, you can try to configure the [default unicast locator list](https://fast-dds.docs.eprosima.com/en/latest/fastdds/transport/listening_locators.html#user-traffic-unicast-locators) with that specific address so the locators announced to the remote participants are exclusively those on your static IP.
